### PR TITLE
feat(3415968370): New deploy transaction

### PR DESCRIPTION
**Pull request by galuzan-starkware**
### Description of the Changes

- Using `starknet` v4
- Passing `as_string` as a flag for returning the compiled contract as a string and not as an object
- Deploy button will be disabled if a user is not logged in OR deploying is in process
- When deploying, the user needs to sign for two transactions, first for `declare` and second for `deploy.` It is all in one single function: `account.declareDeploy`. 

Solves #[3415968370](https://starkware.monday.com/boards/2847513897/pulses/3415968370)

BREAKING CHANGE

---

### Checklist

- [x] Manually tests of the main Application flows are done and passed.
- [ ] New unit / functional tests have been added (whenever applicable).
- [ ] Docs have been updated (whenever relevant).
- [x] PR title is follow the [Conventional Commits](https://www.conventionalcommits.org/) convention: `<type>[optional scope]: <description>`, e.g: `fix: prevent racing of requests`

### CODE REVIEW

File: .env.development [hunk 1]
```diff
@@ -1 +1,4 @@
-REACT_APP_PG_SERVICE_URL=http://localhost:8081/data.json
+REACT_APP_PG_SERVICE_URL=https://vu00h9m96c.execute-api.us-east-2.amazonaws.com/staging/cairo-staging
+#REACT_APP_PG_SERVICE_URL=http://192.168.3.54:8100/data.json
+REACT_APP_STARKNET_ALPHA_SERVICE_URL=https://alpha4.starknet.io
+REACT_APP_SHARP_SERVICE_URL=https://testnet.provingservice.io
```
Comment (dan-ziv) [inline .env.development]: ?

File: package.json [hunk 2]
```diff
@@ -46,7 +46,8 @@
     "@starkware-industries/commons-js-enums": "^1.2.0",
     "@starkware-industries/commons-js-utils": "^1.2.1-dev.1",
     "codemirror": "^5.62.3",
-    "get-starknet": "1.5.0",
+    "get-starknet": "2.0.0",
+    "get-starknet-core": "2.0.0",
```
Comment (dan-ziv) [inline package.json]: ?

File: src/api/playground.js [hunk 3]
```diff
@@ -67,7 +67,12 @@ const prove = async code => {
 };
 
 const compile = async code => {
-  const data = {code, action: ActionTypes.COMPILE_CONTRACT};
+  const data = {
+    code,
+    action: ActionTypes.COMPILE_CONTRACT,
+    compressed: false,
+    as_string: true
```
Comment (dan-ziv) [inline src/api/playground.js:75]: What is this param?

File: src/api/starknet.js [hunk 4]
```diff
@@ -1,28 +1,34 @@
-import {ActionTypes} from '../constants/action-types';
-import {generateRandomHex} from '../utils/generate-random-hex';
-import {apiRequest} from './api-request';
+import {promiseHandler} from '@starkware-industries/commons-js-utils';
+import {getStarknet} from 'get-starknet-core';
 
-const path = process.env.REACT_APP_STARKNET_ALPHA_SERVICE_URL;
+const getAccount = async () => {
+  const {account} = await getStarknet().getLastConnectedWallet();
```
Comment (dan-ziv) [inline src/api/starknet.js:5]: Why do we call `getLastConnectedWallet` and not just `getStarknet().account` ?

**Review by dan-ziv — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (galuzan-starkware) [reply package.json]: `account.declare` and `account.deploy` is supported only on v2 of get-starknet-core.
they break their API, now `getStarknet` is in `get-starknet-core`.

File: src/api/playground.js (hunk 3 above)
Comment (galuzan-starkware) [reply src/api/playground.js:75]: flag for returning the compiled contract as a string and not as an object (wrote it in the description)

File: src/api/starknet.js (hunk 4 above)
Comment (galuzan-starkware) [reply src/api/starknet.js:5]: This is how getStarknet works in version 2

File: .env.development (hunk 1 above)
Comment (galuzan-starkware) [reply .env.development]: Env vars for development...
Is there anything you'd like me to elaborate on?

Comment (dan-ziv) [reply .env.development]: You shouldn't push the manipulated `.env.development`
You can use instead `.env.development.local` which is git ignored.

File: package.json (hunk 2 above)
Comment (dan-ziv) [reply package.json]: Do we need them both?

Comment (galuzan-starkware) [reply package.json]: get-starknet cause it is a peer dependency on commons and get-starknet-core cause we get getStarknet from it

**Review by dan-ziv — APPROVED** [review]

**Pull request merged**